tools/oeutil/host/generate_evidence.cpp

```diff
@@ -952,3 +1284,477 @@ oe_result_t generate_oe_evidence(
+                break;
+            }
+        }
+        else if (
+            strcasecmp(INPUT_PARAM_OPTION_QUOTE_PROC, argv[i]) == 0 ||
+            strcasecmp(SHORT_INPUT_PARAM_OPTION_QUOTE_PROC, argv[i]) == 0)
+        {
+            if (argc < i + 2)
+                break;
+
+            _params.quote_proc = argv[i + 1];
+            if (strcasecmp(INPUT_PARAM_QUOTE_IN_PROC, _params.quote_proc) !=
+                    0 &&
+                strcasecmp(INPUT_PARAM_QUOTE_OUT_OF_PROC, _params.quote_proc) !=
+                    0)
+            {
+                printf(
+                    "Please use 'in' or 'out' with %s.\n",
+                    INPUT_PARAM_OPTION_QUOTE_PROC);
+                return 1;
+            }
+            i += 2;
+        }
+        else if (
+            strcasecmp(INPUT_PARAM_OPTION_ENDORSEMENTS_FILENAME, argv[i]) ==
+                0 ||
+            strcasecmp(
+                SHORT_INPUT_PARAM_OPTION_ENDORSEMENTS_FILENAME, argv[i]) == 0)
+        {
+            if (argc < i + 2)
+                break;
+
+            _params.endorsements_filename = argv[i + 1];
+            i += 2;
+        }
+        else if (
+            strcasecmp(INPUT_PARAM_OPTION_OUT_FILE, argv[i]) == 0 ||
+            strcasecmp(SHORT_INPUT_PARAM_OPTION_OUT_FILE, argv[i]) == 0)
+        {
+            if (argc < i + 2)
+                break;
+
+            _params.out_filename = argv[i + 1];
+            i += 2;
+        }
+        else if (
+            strcasecmp(INPUT_PARAM_OPTION_VERIFY, argv[i]) == 0 ||
+            strcasecmp(SHORT_INPUT_PARAM_OPTION_VERIFY, argv[i]) == 0)
+        {
+            _params.verify = true;
+            i++;
+        }
+        else if (
+            strcasecmp(INPUT_PARAM_OPTION_LOG_FILE, argv[i]) == 0 ||
+            strcasecmp(SHORT_INPUT_PARAM_OPTION_LOG_FILE, argv[i]) == 0)
+        {
+            if (argc < i + 2)
+                break;
+
+            _params.log_filename = argv[i + 1];
+            i += 2;
+        }
+        else if (strcasecmp(INPUT_PARAM_OPTION_VERBOSE, argv[i]) == 0)
+        {
+            _params.verbose = true;
+            i++;
+        }
+        else
+        {
+            printf("Invalid option: %s\n\n", argv[i]);
+            _display_help(argv[0]);
+            return 1;
+        }
+    }
+
+    if (i < argc)
+    {
+        printf("%s has invalid number of parameters.\n\n", argv[i]);
+        _display_help(argv[0]);
+        return 1;
+    }
+
+    if (_params.generate_certificate + _params.generate_legacy_report_remote +
+            _params.generate_sgx_ecdsa + _params.generate_sgx_epid_linkable +
+            _params.generate_sgx_epid_unlinkable !=
+        1)
+    {
+        printf("Please specify to generate a certificate, a report, or "
+               "evidence in SGX_ECDSA, SGX_EPID_LINKABLE or "
+               "SGX_EPID_UNLINKABLE format.\n");
+        return 1;
+    }
+
+    return 0;
+}
+
+oe_result_t _process_params(oe_enclave_t* enclave)
```

Comment:
Avoid abbreviations (like "param").  "args" on the other hand is ok just because C basically standardized it as if it were a word
```suggestion
oe_result_t _process_argss(oe_enclave_t* enclave)
```